### DISCUSSION TITLE
Custom Timeout to the server request made by Client

### DISCUSSION THREAD

**Discussion by ritishbhardwaj** [dennisvang/tufup#136] [Q&A] [answered]
App is installed in client machine and from there the application is making the request to check for updates or download the updates, what is happening in my case is,  when the server was down and in that time the application got stopped or not responsive when making the request. So i got the idea of why not we customize the timeout from the client side. 
Can you tell whether i am right or not and if i am right can you tell me the way how to customize it and where to refer the code.

**Comment by dennisvang** [comment, marked as answer]
>... 
>So i got the idea of why not we customize the timeout from the client side.
>Can you tell whether i am right or not and if i am right can you tell me the way how to customize it and where to refer the code.

Hi @ritishbhardwaj 

Python-tuf defines the [`tuf.ngclient.RequestsFetcher`][1], which has a `socket_timeout` argument.

Tufup defines a *subclass* of `RequestsFetcher`, called [`AuthRequestsFetcher`][2], but we [do not][3] expose this in the tufup client. This was intentional, to limit the complexity for tufup users.

Your best option would probably be to monkey-patch `AuthRequestsFetcher` (or `RequestsFetcher`) to override the default `socket_timeout` value.

Do note that this may have security implications, as discussed in several python-tuf issues. To name a few:

- theupdateframework/python-tuf#2442
- theupdateframework/python-tuf#932
- theupdateframework/python-tuf#204
- ...

[1]: https://github.com/theupdateframework/python-tuf/blob/970ddf9df01ee64df32d1c610502e8d1a1bbdb29/tuf/ngclient/_internal/requests_fetcher.py#L39
[2]: https://github.com/dennisvang/tufup/blob/4bb16ad228b4b6f39e42527cc4367e501d728142/src/tufup/client.py#L335
[3]: https://github.com/dennisvang/tufup/blob/4bb16ad228b4b6f39e42527cc4367e501d728142/src/tufup/client.py#L47

**Reply by ritishbhardwaj** [reply]
I think it is working for me till now for the file of 30MB 
will ping you again if any issue raise again.
Will Solve it together.
